src/contracts/libraries/SlashingAccountingUtils.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+/**
+ * @title Library of utilities for calculations related to slashing in EigenLayer
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ */
+library SlashingAccountingUtils {
+
+    // TODO: set this based on payments epochs
+    uint256 internal constant EPOCH_GENESIS_TIMESTAMP = 1606824000;
+
+    // TODO: define appropriately
+    uint256 internal constant EPOCH_LENGTH_SECONDS = 2 weeks;
+
+    uint256 internal constant SHARE_CONVERSION_SCALE = 1e18;
+
+    // an amount of shares over this will cause overflow when multiplying by `SHARE_CONVERSION_SCALE`
+    uint256 internal constant MAX_VALID_SHARES = type(uint96).max;
+
+    uint256 internal constant BIPS_FACTOR = 10000;
+
+    // TODO: explain this better. basically seems like we may need to set some max factor beyond which shares are just zeroed out
+    uint256 internal constant MAX_SCALING_FACTOR = type(uint256).max / (MAX_VALID_SHARES * SHARE_CONVERSION_SCALE);
+
+    function getEpochFromTimestamp(uint256 timestamp) internal pure returns (int256) {
+        return (int256(timestamp) - int256(EPOCH_GENESIS_TIMESTAMP)) / int256(EPOCH_LENGTH_SECONDS);
+    }
+
+    function currentEpoch() internal view returns (int256) {
+        return getEpochFromTimestamp(block.timestamp);
+    }
+
+    function scaleUp(uint256 shares, uint256 scalingFactor) internal pure returns (uint256) {
```

Comment:
perhaps `normalize` and `denormalize` would be clearer?

Reply:
this is a reasonable suggestion. I would like good naming for these.
perhaps `renormalize` when returning to just 'shares'
`denormalizeShares` and `renormalizeToShares`, or something?

Reply:
works for me

Reply:
+1 `normalize` and `denormalize` something like

```
function normalize(nonNormalizedShares, uint256 scalingFactor) returns (uint256)

function denormalize(normalizedShares, uint256 scalingFactor) returns (uint256)
```

Reply:
renaming done here https://github.com/Layr-Labs/eigenlayer-contracts/pull/543/commits/94a529babf81fe8a108d1354be8cf7155c4d9498, resolving thread